op_builder/xpu/builder.py

```diff
@@ -117,3 +119,247 @@ def jit_load(self, verbose=True):
             os.environ["TORCH_CUDA_ARCH_LIST"] = torch_arch_list
         '''
         return op_module
+
+class SYCLAutoOpBuilder(OpBuilder):
+
+    def builder(self):
+        try:
+            from intel_extension_for_pytorch.xpu.cpp_extension import DPCPPExtension
+        except ImportError:
+            from intel_extension_for_pytorch.xpu.utils import DPCPPExtension
+
+        sycl_include_paths, sycl_sources = self.update_sycl_code_path()
+        print("dpcpp sources = {}".format(sycl_sources))
+        dpcpp_ext = DPCPPExtension(name=self.absolute_name(),
+                                   sources=self.strip_empty_entries(sycl_sources),
+                                   include_dirs=self.strip_empty_entries(sycl_include_paths),
+                                   extra_compile_args={
+                                       'cxx': self.strip_empty_entries(self.cxx_args()),
+                                   },
+                                   extra_link_args=self.strip_empty_entries(self.fixed_aotflags()))
+        return dpcpp_ext
+
+    def sycl_extension(self):
+        if self.is_sycl_enabled():
```

Comment:
This function is very long.  Can we extract three smaller functions?  one for include, one for source.